bangc-ops/kernels/roiaware_pool3d/roiaware_pool3d.cpp

```diff
@@ -232,12 +238,12 @@ mluOpStatus_t MLUOP_WIN_API mluOpRoiawarePool3dForward(
   TENSOR_NUM_CHECK(API, tensor_pts_idx_of_voxels_num, LARGE_TENSOR_NUM, "");
   TENSOR_NUM_CHECK(API, tensor_pooled_features_num, LARGE_TENSOR_NUM, "");
 
-  // check zero element
-  PARAM_CHECK(API, tensor_rois_num > 0);
-  PARAM_CHECK(API, tensor_pts_num > 0);
-  PARAM_CHECK(API, tensor_pts_feature_num > 0);
-  PARAM_CHECK(API, tensor_pts_idx_of_voxels_num > 0);
-  PARAM_CHECK(API, tensor_pooled_features_num > 0);
+  // product of boxes_num and pts_num < 2^31
+  if (boxes_num > INT32_MAX / pts_num) {
```

Comment:
```suggestion
  size_t product_boxes_pts = (size_t)boxes_num * (size_t)pts_num;
  if (product_boxes_pts >= (size_t)INT32_MAX) {
```
你原本的式子无法排除 boxes_num * pts_num = INT32_MAX 的情况，但改为等号后又会因为 floor 特征出错。